WearTilesKotlin/app/src/main/java/com/example/wear/tiles/golden/Workout.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.wear.tiles.golden
+
+import android.content.Context
+import androidx.wear.tiles.ColorBuilders
+import androidx.wear.tiles.DeviceParametersBuilders.DeviceParameters
+import androidx.wear.tiles.ModifiersBuilders.Clickable
+import androidx.wear.tiles.material.Button
+import androidx.wear.tiles.material.ChipColors
+import androidx.wear.tiles.material.CompactChip
+import androidx.wear.tiles.material.Text
+import androidx.wear.tiles.material.TitleChip
+import androidx.wear.tiles.material.Typography
+import androidx.wear.tiles.material.layouts.MultiButtonLayout
+import androidx.wear.tiles.material.layouts.PrimaryLayout
+
+object Workout {
+    const val BUTTON_1_ICON_ID = "1"
+    const val BUTTON_2_ICON_ID = "2"
+    const val BUTTON_3_ICON_ID = "3"
+
+    fun buttonsLayout(
+        context: Context,
+        deviceParameters: DeviceParameters,
+        weekSummary: String,
+        button1Clickable: Clickable,
+        button2Clickable: Clickable,
+        button3Clickable: Clickable,
+        chipClickable: Clickable
+    ) =
+        PrimaryLayout.Builder(deviceParameters)
+            .setPrimaryLabelTextContent(
+                Text.Builder(context, weekSummary)
+                    .setTypography(Typography.TYPOGRAPHY_CAPTION1)
+                    .setColor(ColorBuilders.argb(GoldenTilesColors.Blue))
+                    .build()
+            )
+            .setContent(
+                MultiButtonLayout.Builder()
+                    .addButtonContent(
+                        Button.Builder(context, button1Clickable).setIconContent(BUTTON_1_ICON_ID)
+                            .build()
+                    )
+                    .addButtonContent(
+                        Button.Builder(context, button2Clickable).setIconContent(BUTTON_2_ICON_ID)
+                            .build()
+                    )
+                    .addButtonContent(
+                        Button.Builder(context, button3Clickable).setIconContent(BUTTON_3_ICON_ID)
+                            .build()
+                    )
+                    .build()
+            )
+            .setPrimaryChipContent(
+                CompactChip.Builder(context, "More", chipClickable, deviceParameters)
+                    .setChipColors(
+                        ChipColors(
```

Comment:
bordering on the don't review for content, but would it be helpful to implement these designs using themes using Colors? then using primary or secondary ChipColors factories?

Reply:
I added a comment for this elsewhere. I'm not sure themes for tiles is worth it - as I was implementing all of these, I found it easier to specify colors explicitly.

In a real project, I imagine it'll be even less worth it:

- a real project won't have many tiles
- themes aren't directly reusable between Compose/this
- even if they were reusable, it's not guaranteed that we'd want the same combination of colors
- tiles (/wear?) is always dark mode and doesn't support multi-theming

I'm not sure what it provides.

Reply:
The link you provided here to an example project does help show some utility - https://github.com/yschimke/rememberwear/blob/main/wear/src/main/kotlin/com/google/wear/soyted/tile/RememberWearTileRenderer.kt#L116

---

There's an example of a theme being created in `ComponentAudit.kt` which we can leave as-is / improve. 

I think the Golden Tiles don't benefit from themes because it'll only serve to showcase the overhead without any of the benefit because it's not being shared anywhere (vs. your RememberTheMilk project which was sharing it with the main app).

Reply:
IF we want people to be able to take these tiles and copy and paste them, I think it's useful. Or maybe a library of reference tile designs, "templates" that we provide.  But your call.

Reply:
Let's leave it for now and we'll revisit if there's feedback. I think these samples should facilitate understanding and we shouldn't necessarily be encouraging copy and pasting.